src/stores/pacbioLibraries.js

```diff
@@ -107,25 +54,26 @@ export const usePacbioLibrariesStore = defineStore('pacbioLibraries', {
      * @returns {Array<Object>} - An array of library objects, each with id, tag_group_id, sample_name, barcode, and other attributes.
      */
     librariesArray: (state) => {
-      return Object.values(state.libraryState.libraries)
+      const pacbioRootStore = usePacbioRootStore()
+      return Object.values(state.libraries)
         .filter((library) => library.tube)
         .map((library) => {
           const { id, request, tag_id, tag, tube, ...attributes } = library
           const tagId = tag_id ?? tag
-          /*Get the tag group ID from the library's tag ID or the tag's group ID. Why this is required?
+
+          /*Get the tag group ID from the library's tag ID or from the tag in pacbioRoot store(where all pacbio tags are kept). Why is this required?
           The librariesArray is called in multiple places (in create and edit context) to get the libraries. 
           Therefore, librariesArray needs to search for the tag first in libraryTags. 
-          If not found, it should then look for it in tagState->tags. 
-          It's important that tagState->tags will only get populated if a fetchPacbioTagSets is called before, 
+          If not found, it should then look for it in 'pacbioRoot' store tags. 
+          It's important to note that 'pacbioRoot' store tags will only get populated if a 'pacbioRoot' store action fetchPacbioTagSets is called before, 
           which may not happen in all the places where it's called. 
           Hence, a search in both places is required to ensure that librariesArray returns the correct tag 
           associated with all libraries."*/
-
           const tagGroupId = tagId
-            ? state.libraryState.libraryTags[tagId]
-              ? state.libraryState.libraryTags[tagId].group_id
-              : state.tagState.tags[tagId]
-                ? state.tagState.tags[tagId].group_id
+            ? state.libraryTags[tagId]
```

Comment:
Do you need to use a nested ternary here? Can you do the question mark after [tagId] ...

Reply:
Good point, the outer one can very well be removed as we are checking state.libraryTags[tagId].